src/pykiso/lib/auxiliaries/zephyr.py

```diff
@@ -0,0 +1,212 @@
+        )
+        test_name = test_name if test_name is not None else self.test_name
+        if test_directory is None:
+            raise ZephyrError("test_directory parameter is not set.")
+        if test_name is None:
+            raise ZephyrError("test_name parameter is not set.")
+
+        if self.running:
+            raise ZephyrError("Twister test is already running")
+
+        self.outdir = str(Path(test_directory).resolve() / "twister-out")
+        logging.internal_info(f"Using Twister output directory: {self.outdir}")
+        self.channel.cc_send(
+            {
+                "command": "start",
+                "args": [
+                    "-vv",
+                    "-T",
+                    test_directory,
+                    "-s",
+                    test_name,
+                    "--outdir",
+                    self.outdir,
+                    "--report-dir",
+                    self.outdir,
+                ],
+            }
+        )
+
+        self.running = True
+        if self.wait_for_start:
+            # Read twister output until test case was started
+            while True:
+                line = self._readline()
+                if line is None:
+                    return
+
+                if "OUTPUT: START - " in line:
+                    log.internal_debug("Zephyr test started.")
+                    return
+
+    def _readline(self) -> Optional[str]:
+        """Read a line from the twister process
+
+        :return: The line as string or None if the process has finished
+        """
+        if not self.running:
+            return None
+
+        while True:
+            msg = self.channel.cc_receive(timeout=None)["msg"]
+            if msg is None:
+                return None
+            line = msg.get("stderr")
+            if line is None:
+                if "exit" in msg:
+                    self.running = False
+                    return None
+
+            # Twister outputs many lines without a real message, remove those
+            test_output_pattern = "- OUTPUT:"
+            output = line.find(test_output_pattern)
+            if output != -1:
+                message = line[output + len(test_output_pattern) :]
+                if len(message) == 0 or message.isspace():
+                    continue
+            log.internal_debug(f"Twister: {line}")
+            return line
+
+    def wait_test(self) -> TestResult:
+        """Wait for the test to finish
+
+        :return: The result of the test, taken from the xunit output of twister
+        """
+        if not self.running:
+            raise ZephyrError("Twister was not started so can not wait.")
+
+        # Read the twister console for log output and wait for exit
+        while True:
+            line = self._readline()
+            if line is None:
+                break
+            # Look for relevant messages in the output
+            if "OUTPUT:  PASS - " in line:
```

Comment:
Same here you could create 2 class variables 
`TEST_RESULT_PATTERN = "OUTPUT:  "`
`TEST_PASSED_PATTERN = "PASS - "`

-> `if self.TEST_RESULT_PATTERN + self.TEST_PASSED_PATTERN in line:`

Reply:
Done